actors/builtin/miner/miner_actor.go

```diff
@@ -2067,6 +2068,297 @@ func (a Actor) RepayDebt(rt Runtime, _ *abi.EmptyValue) *abi.EmptyValue {
+
+	store := adt.AsStore(rt)
+	var stReadOnly State
+	rt.StateReadonly(&stReadOnly)
+	info := getMinerInfo(rt, &stReadOnly)
+
+	rt.ValidateImmediateCallerIs(append(info.ControlAddresses, info.Owner, info.Worker)...)
+
+	sectors, err := LoadSectors(store, stReadOnly.Sectors)
+	builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to load sectors array")
+
+	powerDelta := NewPowerPairZero()
+	pledgeDelta := big.Zero()
+
+	type updateAndSectorInfo struct {
+		update     *ReplicaUpdate
+		sectorInfo *SectorOnChainInfo
+	}
+
+	var sectorsDeals []market.SectorDeals
+	var sectorsDataSpec []*market.SectorDataSpec
+	var validatedUpdates []*updateAndSectorInfo
+	sectorNumbers := bitfield.New()
+	for i := range params.Updates {
+		update := params.Updates[i]
+		// Bitfied.IsSet() is fast when there are only locally-set values.
+		set, err := sectorNumbers.IsSet(uint64(update.SectorID))
+		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "error checking sector number")
+		if set {
+			rt.Log(rtt.INFO, "duplicate sector being updated %d, skipping", update.SectorID)
+			continue
+		}
+
+		sectorNumbers.Set(uint64(update.SectorID))
+
+		if len(update.ReplicaProof) > 4096 {
+			rt.Log(rtt.INFO, "update proof is too large (%d), skipping sector %d", len(update.ReplicaProof), update.SectorID)
+			continue
+		}
+
+		if len(update.Deals) <= 0 {
+			rt.Log(rtt.INFO, "must have deals to update, skipping sector %d", update.SectorID)
+			continue
+		}
+
+		if uint64(len(update.Deals)) >= SectorDealsMax(info.SectorSize) {
+			rt.Log(rtt.INFO, "more deals than policy allows, skipping sector %d", update.SectorID)
+			continue
+		}
+
+		if update.Deadline >= WPoStPeriodDeadlines {
+			rt.Log(rtt.INFO, "deadline %d not in range 0..%d, skipping sector %d", update.Deadline, WPoStPeriodDeadlines, update.SectorID)
+			continue
+		}
+
+		if !update.NewSealedSectorCID.Defined() {
+			rt.Abortf(exitcode.ErrIllegalArgument, "new sealed CID undefined")
+		}
+
+		if update.NewSealedSectorCID.Prefix() != SealedCIDPrefix {
+			rt.Abortf(exitcode.ErrIllegalArgument, "new sealed CID had wrong prefix")
+		}
+
+		err = stReadOnly.CheckSectorHealthExcludeUnproven(store, update.Deadline, update.Partition, update.SectorID)
+		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalArgument, "sector is not healthy")
+
+		sectorInfo, err := sectors.MustGet(update.SectorID)
+		if err != nil {
+			rt.Log(rtt.INFO, "failed to get sector, skipping sector %d", update.SectorID)
+			continue
+		}
+
+		if len(sectorInfo.DealIDs) != 0 {
+			rt.Log(rtt.INFO, "cannot update sector with deals, skipping sector %d", update.SectorID)
+			continue
+		}
+
+		code := rt.Send(
+			builtin.StorageMarketActorAddr,
+			builtin.MethodsMarket.ActivateDeals,
+			&market.ActivateDealsParams{
+				DealIDs:      update.Deals,
+				SectorExpiry: sectorInfo.Expiration,
+			},
+			abi.NewTokenAmount(0),
+			&builtin.Discard{},
+		)
+
+		if code != exitcode.Ok {
```

Comment:
well.. annoying me again - but like...should we consider active the valid deals instead of skipping the whole sector? (similar to what was discussed for prove commit back then for  chocolate )

Reply:
Would be nice, but the Market Actor doesn't tell you which deals were successfully activated and which ones were failures, so it's impossible to selectively skip.